test/e2e/main_test.go

```diff
@@ -1,10 +1,60 @@
 package e2e_test
 
 import (
+	"fmt"
 	"os"
 	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/machine-config-operator/cmd/common"
 )
 
 func TestMain(m *testing.M) {
+	cb, err := common.NewClientBuilder("")
+	if err != nil {
+		panic(err)
+	}
+	cClient := cb.ConfigClientOrDie("bootstrap-checks")
+
+	clusterVersion, err := cClient.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+	if err != nil {
+		panic(err)
+	}
+	mcOperator, err := cClient.ConfigV1().ClusterOperators().Get("machine-config", metav1.GetOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	var mcoFailingErr error
+	for _, cond := range mcOperator.Status.Conditions {
+		if cond.Type == configv1.OperatorFailing && cond.Status == configv1.ConditionTrue {
+			mcoFailingErr = fmt.Errorf("MCO is failing with %q", cond.Message)
+			fmt.Println(mcoFailingErr)
+			break
+		}
+	}
+
+	// make sure the CVO is also failing...
+	var cvoFailingErr error
+	for _, cond := range clusterVersion.Status.Conditions {
+		if cond.Type == configv1.OperatorFailing && cond.Status == configv1.ConditionTrue {
+			cvoFailingErr = fmt.Errorf("CVO is failing with %q", cond.Message)
+			fmt.Println(cvoFailingErr)
+			break
+		}
+	}
+
+	if mcoFailingErr != nil && cvoFailingErr == nil {
+		panic("CVO is not failing but the MCO is failing, something went wrong")
```

Comment:
In theory this is racy since the MCO could start failing right when we check, but...eh.  Not a real world concern I think.

Reply:
yeah, I'm adding this gate here just to be sure we are cool with 1) CVO being good 2) MCO being good just _before_ kicking e2e, but yeah, we could theoretically start failing afterwards